### ISSUE TITLE
InvalidComparison error from integrate

### ISSUE THREAD

**Issue by oscarbenjamin** [sympy/sympy#17473]
From the mailing list:
https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/sympy/fpRIf9c79zw/xYhYha8cDgAJ

```
 integrate(sin(x**n),x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\SymPyWorkbook\lib\site-packages\sympy\integrals\integrals.py", line 1477, in integrate
    return integral.doit(**doit_flags)
  File "C:\SymPyWorkbook\lib\site-packages\sympy\integrals\integrals.py", line 541, in doit
    function, xab[0], **eval_kwargs)
  File "C:\SymPyWorkbook\lib\site-packages\sympy\integrals\integrals.py", line 1012, in _eval_integral
    h = meijerint_indefinite(g, x)
  File "C:\SymPyWorkbook\lib\site-packages\sympy\integrals\meijerint.py", line 1621, in meijerint_indefinite
    res = _meijerint_indefinite_1(f.subs(x, x + a), x)
  File "C:\SymPyWorkbook\lib\site-packages\sympy\integrals\meijerint.py", line 1684, in _meijerint_indefinite_1
    if b < 0 or f.subs(x, 0).has(nan, zoo):
  File "C:\SymPyWorkbook\lib\site-packages\sympy\core\relational.py", line 304, in __nonzero__
    raise TypeError("cannot determine truth value of Relational")
TypeError: cannot determine truth value of Relational
```
David: I suppose I expected it to return an unevaluated integral -  integral(sin(x**n),x)

@jksuom:
```
We should have, for example, `if (b < 0) == True or ...` here
  File "C:\SymPyWorkbook\lib\site-packages\sympy\integrals\meijerint.py", line 1684, in _meijerint_indefinite_1
    if b < 0 or f.subs(x, 0).has(nan, zoo):
```

**Comment by oscargus** [comment]
I guess `b.is_negative` is also an alternative?

**Comment by jksuom** [comment]
I think that `b` would be a number in typical cases. `b.is_negative` would then work as well though it would take a little more time.

**Comment by oscargus** [comment]
I thought I recognized the code...

This is already fixed (but not merged) in #17343.
